symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/getAddonDescriptionAPIAction.class.php

```diff
@@ -24,37 +24,43 @@ class getAddonDescriptionAPIAction extends baseAddonAction
 {
     /**
      * @param sfRequest $request
-     * @return mixed
+     * @return mixed|string
      * @throws CoreServiceException
-     * @throws sfStopException
+     * return string is base 64 encoded html which is coming from MP backend
+     * 0 indicate Marketplace network error
+     * 1 indicate Marketplace middlewere down.
      */
     public function execute($request)
     {
         $data = $request->getParameterHolder()->getAll();
         $addonId = $data['addonID'];
-        $addonList = $this->getAddons();
-        foreach ($addonList as $addon) {
-            if ($addon['id'] == $addonId) {
-                $addonDesCriptionURL = $addon['links']['desc'];
+        try {
+            $addonList = $this->getAddons();
+            foreach ($addonList as $addon) {
+                if ($addon['id'] == $addonId) {
+                    $addonDesCriptionURL = $addon['links']['desc'];
+                }
             }
+            $addonDescription = $this->getDescription($addonDesCriptionURL);
+            $this->addonDescription = base64_decode($addonDescription);
+        } catch (GuzzleHttp\Exception\ConnectException $e) {
+            echo json_encode('0');
```

Comment:
Are we showing "No internet connection" msg based on this error?

Reply:
yes

Reply:
I add it in doc comment